File: backend/app/schemas/tracking.py
import uuid
from typing import Optional
from pydantic import BaseModel


class GPSPingRequest(BaseModel):
    latitude: float
    longitude: float
    eta_minutes: Optional[int] = None


class TrackingResponse(BaseModel):
    delivery_id: uuid.UUID
    current_latitude: Optional[float] = None
    current_longitude: Optional[float] = None
    current_status: str
    eta_minutes: Optional[int] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/tracking.ts
export interface TrackingSnapshot {
  delivery_id: string;
  current_latitude?: number | null;
  current_longitude?: number | null;
  current_status: string;
  eta_minutes?: number | null;
}

export interface OrderTrackingResponse {
  order_id: string;
  status: string;
  tracking: TrackingSnapshot | null;
}
